circuit/src/contracts-registry/src/benchmarking.rs

```diff
@@ -0,0 +1,250 @@
+            test_contract_1.clone(),
+        );
+
+        let contract_id_2 = test_contract_2.generate_id::<T>();
+        crate::ContractsRegistry::<T>::insert(
+            test_contract_2.generate_id::<T>(),
+            test_contract_2.clone(),
+        );
+    }: _(RawOrigin::Root, requester_1.clone(), contract_id_1.clone())
+    verify {
+        assert_ok!(ContractsRegistry::<T>::purge(RawOrigin::Root.into(), requester_2, contract_id_2));
+    }
+
+    fetch_contracts {
+        let test_contract_author_1: RegistryContract<T::Hash, T::AccountId, BalanceOf<T>, T::BlockNumber> = RegistryContract {
+            code_txt: b"some_code_1".to_vec(),
+            bytes: vec![],
+            author: account("TEST", 1_u32, USER_SEED),
+            author_fees_per_single_use: None,
+            abi: None,
+            action_descriptions: vec![],
+            info: None,
+            meta: ContractMetadata::new(vec![], vec![], vec![], vec![], Some(b"contract 1".to_vec()), None, None, None, None),
+        };
+        let test_contract_author_2: RegistryContract<T::Hash, T::AccountId, BalanceOf<T>, T::BlockNumber> = RegistryContract {
+            code_txt: b"some_code_2".to_vec(),
+            bytes: vec![],
+            author: account("TEST", 1_u32, USER_SEED),
+            author_fees_per_single_use: None,
+            abi: None,
+            action_descriptions: vec![],
+            info: None,
+            meta: ContractMetadata::new(vec![], vec![], vec![], vec![], Some(b"contract 2".to_vec()), None, None, None, None),
+        };
+        let test_contract_author_3: RegistryContract<T::Hash, T::AccountId, BalanceOf<T>, T::BlockNumber> = RegistryContract {
+            code_txt: b"some_code_3".to_vec(),
+            bytes: vec![],
+            author: account("TEST", 2_u32, USER_SEED),
+            author_fees_per_single_use: None,
+            abi: None,
+            action_descriptions: vec![],
+            info: None,
+            meta: ContractMetadata::new(vec![], vec![], vec![], vec![], None, None, None, None, None),
+        };
+
+            crate::ContractsRegistry::<T>::insert(
+                test_contract_author_1.generate_id::<T>(),
+                test_contract_author_1.clone(),
+            );
+            crate::ContractsRegistry::<T>::insert(
+                test_contract_author_2.generate_id::<T>(),
+                test_contract_author_2.clone(),
+            );
+            crate::ContractsRegistry::<T>::insert(
+                test_contract_author_3.generate_id::<T>(),
+                test_contract_author_3.clone(),
+            );
+
+        let author_1: T::AccountId = account("TEST", 1_u32, USER_SEED);
+
+    }: {ContractsRegistry::<T>::fetch_contracts(Some(author_1.clone()), Some(b"contract".to_vec()))}
+    verify {
+        assert_eq!(ContractsRegistry::<T>::fetch_contracts(Some(author_1), Some(b"contract".to_vec())), Ok(vec![test_contract_author_1.clone(), test_contract_author_2.clone()]));
```

Comment:
Suggestion : 
@MaciejBaj Do we want to follow the lint rule where in we don't allow a line to be more than 80 characters wide?
Some teams tend to follow that. Purely comes down to preference though.

Reply:
rely on cargo fmt there

Reply:
this one does seem too long though, perhaps this file wasn't formatted?